src/ScheduledTasks/Models/ScheduleConfiguration.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.DurableTask.ScheduledTasks;
+
+/// <summary>
+/// Configuration for a scheduled task.
+/// </summary>
+class ScheduleConfiguration
+{
+    string orchestrationName;
+    TimeSpan interval;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ScheduleConfiguration"/> class.
+    /// </summary>
+    /// <param name="scheduleId">The ID of the schedule.</param>
+    /// <param name="orchestrationName">The name of the orchestration to schedule.</param>
+    /// <param name="interval">The interval between schedule executions. Must be positive and at least 1 second.</param>
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring as nullable.
+    public ScheduleConfiguration(string scheduleId, string orchestrationName, TimeSpan interval)
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring as nullable.
+    {
+        this.ScheduleId = Check.NotNullOrEmpty(scheduleId, nameof(scheduleId));
+        this.OrchestrationName = Check.NotNullOrEmpty(orchestrationName, nameof(orchestrationName));
```

Comment:
if you change this to `this.orchestrationName = ...`  then you can remove the `#pragma warning disable`